### PR TITLE
Convert physics.f90 to Python

### PR DESCRIPTION
**Pull request by timothy-nunn**
## Description

Converts the remaining physics.f90 module into Python

## Checklist

I confirm that I have completed the following checks:

- [x] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [x] I have added new tests where appropriate for the changes I have made.
- [x] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [x] If I have made documentation changes, I have checked they render correctly.
- [x] I have added documentation for my change, if appropriate.

### CODE REVIEW

**Comment by timothy-nunn** [comment]
When converting `bootstrap_fraction_nevins` to Python, I have replaced the `quanc8` integrator with the scipy integrator. This is because...
1. We cannot use the Fortran solver from Python (it requires a callable)
2. It is ideal to use industry-standard solvers where possible

This causes the unit test, created from values seen in `large-tokamak`, to change from `886436144167.0579` to `889258771342.7881`. @chris-ashe do you have any intuition as to how significant this change is?

File: tests/regression/scenarios/large-tokamak/ref.MFILE.DAT [hunk 1]
```diff
@@ -460,7 +459,7 @@
  Injection_power_to_electrons_(MW)_______________________________________ (pinjemw)_____________________      7.5398E+01 OP 
  Ignited_plasma_switch_(0=not_ignited,_1=ignited)________________________ (ignite)______________________              0    
  Power_into_divertor_zone_via_charged_particles_(MW)_____________________ (pdivt)_______________________      1.5710E+02 OP 
- Psep_/_R_ratio_(MW/m)___________________________________________________ (pdivt/rmajor)________________      1.9637E+01 OP 
+ Psep_/_R_ratio_(MW/m)___________________________________________________ (pdivt/physics_variables.rmajo      1.9637E+01 OP
```
Comment (jonmaddock) [inline tests/regression/scenarios/large-tokamak/ref.MFILE.DAT:462]: This has got chopped off.

File: tests/regression/scenarios/large-tokamak/ref.MFILE.DAT [hunk 2]
```diff
@@ -546,7 +545,7 @@
  Pfirsch-Schlueter_fraction______________________________________________ (psipf)_______________________      0.0000E+00    
  Auxiliary_current_drive_fraction________________________________________ (faccd)_______________________      1.1139E-03    
  Inductive_fraction______________________________________________________ (facoh)_______________________      5.8003E-01    
- Total___________________________________________________________________ (plasipf+faccd+facoh)_________      1.0000E+00    
+ Total___________________________________________________________________ (current_drive_variables.plasi      1.0000E+00
```
Comment (jonmaddock) [inline tests/regression/scenarios/large-tokamak/ref.MFILE.DAT:548]: Chopped off.

**Review by jonmaddock — COMMENTED** [review]
Looks great Tim, huge one! Just a couple of chopped-off descriptions in the mfile output, but otherwise fine.

**Review by jonmaddock — APPROVED** [review]

**Comment by jonmaddock** [comment]
Can't rebase due to conflicts. Can you sort these, then good to go in.

**Pull request merged**